dev-tools/phpstan/src/Extension/PregMatchParameterOutExtension.php

```diff
@@ -60,9 +60,9 @@ public function getParameterOutTypeFromStaticMethodCall(MethodReflection $method
             $flagsType = $scope->getType($flagsArg->value);
         }
 
-        if ($methodReflection->getName() === 'match') {
-            return $this->regexShapeMatcher->matchExpr($patternArg->value, $flagsType, TrinaryLogic::createMaybe(), $scope);
-        }
-        return $this->regexShapeMatcher->matchAllExpr($patternArg->value, $flagsType, TrinaryLogic::createMaybe(), $scope);
+        $matcherMethod = ('match' === $methodReflection->getName()) ? 'matchExpr' : 'matchAllExpr';
+
+        // @phpstan-ignore method.dynamicName
+        return $this->regexShapeMatcher->{$matcherMethod}($patternArg->value, $flagsType, TrinaryLogic::createMaybe(), $scope);
```

Comment:
I would try to get arround dynamic method calls at all costs, but if you prefer why not.

Reply:
I don't like dynamic calls either, but in this case we have closed set of methods to call, both with the same params. I find this more readable than if/else when you need to analyse separate calls.

Reply:
@keradus @kubawerlos any thought from you guys? If there's no strong "no" I would like to merge this to prevent adding new entries to baseline like [here](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8106#discussion_r1764677644).

Reply:
We will migrate it to `$this->method(...)` when we drop support for PHP < 8.1 anyways, right? If so, let's go with this dynamic call, especially since we have it in other places.